compiler/rustc_metadata/src/creader.rs

```diff
@@ -758,6 +758,27 @@ impl<'a, 'tcx> CrateLoader<'a, 'tcx> {
         self.inject_dependency_if(cnum, "a panic runtime", &|data| data.needs_panic_runtime());
     }
 
+    fn inject_compiler_builtins(&mut self, krate: &ast::Crate) {
+        // Don't inject compiler_builtins for core and compiler_builtins.
+        if attr::contains_name(&krate.attrs, sym::no_core)
+            || attr::contains_name(&krate.attrs, sym::compiler_builtins)
+        {
+            return;
+        }
+
+        info!("loading compiler_builtins");
+
+        let Some(cnum) = self.resolve_crate(sym::compiler_builtins, DUMMY_SP, CrateDepKind::Implicit) else {
```

Comment:
```suggestion
        let Some(cnum) = self.resolve_crate(sym::compiler_builtins, DUMMY_SP, CrateDepKind::Explicit) else {
```
`extern crate` item from code will result in a call like this (see `fn process_extern_crate`).
So this change should make the behavior closer to whatever we've been doing previously, maybe it will fix the issues.

Reply:
This doesn't help.